app/src/main/java/com/humberto/tasky/auth/presentation/registration/RegisterViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.humberto.tasky.auth.presentation.registration
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.humberto.tasky.auth.domain.UserDataValidator
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class RegisterViewModel @Inject constructor(
+    private val userDataValidator: UserDataValidator
+): ViewModel() {
+
+    var state by mutableStateOf(RegisterState())
+        private set
+
+    private val eventChannel = Channel<RegisterEvent>()
+    val events = eventChannel.receiveAsFlow()
+
+    init {
+        viewModelScope.launch {
+            collectCombinedEmailAndPassword()
+        }
+    }
+
+    private fun collectCombinedEmailAndPassword() {
+        combine(
+            snapshotFlow { state.fullName.text },
+            snapshotFlow { state.email.text },
+            snapshotFlow { state.password.text }
+        ) { fullName, email, password ->
+            val isValidFullName = userDataValidator.isValidFullName(fullName.toString())
+            val isValidEmail = userDataValidator.isValidEmail(email.toString())
+            val passwordValidationState =
+                userDataValidator.validatePassword(password.toString())
+            state = state.copy(
+                isFullNameValid = isValidFullName,
+                isEmailValid = isValidEmail,
+                passwordValidationState = passwordValidationState,
+                canRegister = isValidEmail &&
+                        !state.isRegisteringIn &&
+                        passwordValidationState.isValidPassword
+            )
+        }.launchIn(viewModelScope)
```

Comment:
You could've put this logic in 3 separate flows rather than combining them all into one. The texts will always just change one at a time and validation of one text field also does not rely on the validation of another, so putting these in separate flow collectors would only validate what needs to be validated. Right now, every keystroke in the email field will also validate full name and password again.